python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1335,6 +1335,34 @@ def _get_proxy_base_class(cls):
     return object
 
 
+def as_proxy_object(obj: Any) -> Any:
+    """
+    Wraps a cudf or pandas object in a proxy object if applicable.
+
+    There will be no memory transfer, i.e., GPU objects stay on GPU and
+    CPU objects stay on CPU. The object will be wrapped in a
+    proxy object. This is useful for ensuring that the object is
+    compatible with the fast-slow proxy system.
+
+    Parameters
+    ----------
+    obj : Any
+        The object to wrap.
+
+    Returns
+    -------
+    Any
+        The wrapped proxy object if applicable, otherwise the original object.
+    """
+    if _is_final_type(obj):
+        typ = get_final_type_map()[type(obj)]
+        return typ._fsproxy_wrap(obj, None)
+    elif _is_intermediate_type(obj):
+        typ = get_intermediate_type_map()[type(obj)]
+        return typ._fsproxy_wrap(obj, None)
```

Comment:
Does your `test_as_proxy_object` test this case currently (I think the caller would want to be wrapping the result of e.g `pandas.Series(...).dt`)?

This case seems a bit ill-defined because the second argument here normally should be `(constructor, args, kwargs)` which we don't have IIUC, so `None` might cause some issues if this result was used further.

If I'm wrong, it would be good to add a test for this. Otherwise, I think it would be good to raise a `NotImplementedError` for now since the cuml use case is solved here.